patroni/dcs/__init__.py

```diff
@@ -161,13 +171,13 @@ def conn_url(self):
             self.data['conn_url'] = conn_url
             return conn_url
 
-    def conn_kwargs(self, auth=None):
+    def conn_kwargs(self, auth: Union[Any, Dict[str, Any], None] = None) -> Dict[str, Any]:
```

Comment:
```suggestion
    def conn_kwargs(self, auth: Union[Dict[str, Any], None] = None) -> Dict[str, Any]:
```

It apparently handles only `dict` and `None`.

Reply:
It does, but `postgresql.authentication.{superuser,replication,rewind}` could contain something different from dict and there are no external checks for that.
The one that we have is located down this method: `if auth and isinstance(auth, dict):`.
If we remove Any from Union pyright starts complaining.